test/test_transforms.py

```diff
@@ -509,5 +510,130 @@ def test_scriptmodule_TimeMasking(self):
+            win_length=n_fft, center=True, window='hann', n_mels=n_mels, htk=True, norm=None)
+        # Note: Using relaxed rtol instead of atol
+        assert torch.allclose(melspec_ta, torch.from_numpy(melspec_lr[None, ...]), rtol=1e-3)
+
+    @unittest.skipIf(not IMPORT_LIBROSA, 'Librosa is not available')
+    def test_InverseMelScale(self):
+        """InverseMelScale transform is comparable to that of librosa"""
+        n_fft = 2048
+        n_mels = 256
+        n_stft = n_fft // 2 + 1
+        hop_length = n_fft // 4
+
+        # Prepare mel spectrogram input. We use torchaudio to compute one.
+        sound, sample_rate = self._get_sample_data(
+            'steam-train-whistle-daniel_simon.wav', offset=2**10, num_frames=2**14)
+        spec_orig = F.spectrogram(
+            sound, pad=0, window=torch.hann_window(n_fft), n_fft=n_fft,
+            hop_length=hop_length, win_length=n_fft, power=2, normalized=False)
+        melspec_ta = transforms.MelScale(n_mels=n_mels, sample_rate=sample_rate)(spec_orig)
+        melspec_lr = melspec_ta.cpu().numpy().squeeze()
+        # Perform InverseMelScale with torch audio and librosa
+        spec_ta = transforms.InverseMelScale(
+            n_stft, n_mels=n_mels, sample_rate=sample_rate)(melspec_ta)
+        spec_lr = librosa.feature.inverse.mel_to_stft(
+            melspec_lr, sr=sample_rate, n_fft=n_fft, power=2.0, htk=True, norm=None)
+        spec_lr = torch.from_numpy(spec_lr[None, ...])
+
+        # Align dimensions
+        # librosa does not return power spectrogram
+        spec_orig = spec_orig.sqrt()
+        spec_ta = spec_ta.sqrt()
+
+        # For debug
+        # _plot_melspecs(sample_rate, spec_orig, spec_ta, spec_lr)
+
+        # The spectrograms reconstructed by librosa and torchaudio are not very comparable elementwise.
+        threshold = 2.5
+        # This is because they use different approximation algorithms and resulting values can live
+        # in different magniude.
+        # See https://github.com/pytorch/audio/pull/366 for the discussion of the choice of algorithm
+        # To get this number do the following and pick something bigger than that;
+        # print('p1 dist (  lr <-> ta):', torch.dist(spec_lr, spec_ta, p=float('inf')))
+        assert torch.allclose(spec_ta, spec_lr, atol=threshold)
+
+        threshold = 1500.0
+        # This threshold was choosen empirically, based on the following observations
+        # print('p1 dist (orig <-> ta):', torch.dist(spec_orig, spec_ta, p=1))
+        # print('p1 dist (orig <-> lr):', torch.dist(spec_orig, spec_lr, p=1))
+        # print('p1 dist (  lr <-> ta):', torch.dist(spec_lr, spec_ta, p=1))
+        # >>> p1 dist (orig <-> ta): tensor(1482.1917)
+        # >>> p1 dist (orig <-> lr): tensor(1420.7103)
+        # >>> p1 dist (  lr <-> ta): tensor(881.7889)
+        assert torch.dist(spec_orig, spec_ta, p=1) < threshold
+
+
+def _plot_melspecs(sample_rate, spec_original, spec_ta, spec_lr):
```

Comment:
Thanks for providing the code, but can you put this in a comment on the PR instead of in here?

Reply:
That being said, please feel free to show the graph in the PR comment :)